File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/core/Module.java
package app.zxtune.core;

import app.zxtune.Releaseable;
import app.zxtune.TimeStamp;

/**
 * Module interface
 */
public interface Module extends PropertiesAccessor, AdditionalFiles, Releaseable {

  /**
   * @return Module's duration in milliseconds
   */
  TimeStamp getDuration();

  /**
   * Creates new player object
   *
   * @throws RuntimeException in case of error
   */
  Player createPlayer(int samplerate);
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/core/Module.kt
package app.zxtune.core

import app.zxtune.Releaseable
import app.zxtune.TimeStamp

/**
 * Module interface
 */
interface Module : PropertiesAccessor, AdditionalFiles, Releaseable {
  /**
   * @return Module's duration in milliseconds
   */
  val duration: TimeStamp

  /**
   * Creates new player object
   *
   * @throws RuntimeException in case of error
   */
  fun createPlayer(samplerate: Int): Player
}
